### ISSUE TITLE
Yield! in monad builder

### ISSUE THREAD

**Issue by 3Rafal** [fsprojects/FSharpPlus#386]
Currently, F#+ `monad.plus` builder doesn't support `yield!`. 
I think it may be more intuitive to support it, as it would make `monad.plus` similar to FSharp.Core sequence expression.

```fsharp
let l1 = [1;2;3]
let l2 = [4;5;6]

// F# list comp. expr.
let lConcat = [
    yield! l1
    yield! l2 ]

// This doesn't work currently
let lConcat': int list = monad.plus {
    yield! l1
    yield! l2 }

// This works fine
let lConcat'': int list = monad.plus {
    return! l1
    return! l2 }
```

Above example will work correctly if we create `Builder.YieldFrom` method that is an alias to `Builder.ReturnFrom`

**Comment by gusty** [comment]
It would actually be an alias of `return!`. I didn't define it in order to reserve it for future use, but so far I didn't came up with an interesting definition other than that.

I think at this point we could add it as an alias of `return!` for the reasons you exposed.

Does anyone have a strong opinion for/against doing this?

**Comment by 3Rafal** [comment]
> It would actually be an alias of return!

@gusty, yes. What I wanted to say is that `Builder.YieldFrom` will have the same implementation as `Builder.ReturnFrom`. That will make `yield!` an alias for `return!`.

**Comment by 3Rafal** [comment]
Closed by #410
